assets/src/edit-story/components/autoSaveHandler/index.js

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useEffect, useState } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { useConfig, useHistory, useStory } from '../../app';
+
+function AutoSaveHandler() {
+  const { autoSaveInterval } = useConfig();
+  const {
+    state: { hasNewChanges },
+  } = useHistory();
+  const {
+    state: {
+      story: { status },
+    },
+    actions: { autoSave, saveStory },
+  } = useStory();
+
+  const [didAutoSave, setDidAutoSave] = useState(false);
+  const isDraft = 'draft' === status;
+
+  // If autoSaveInterval is set to 0 or not defined, don't.
+  if (!autoSaveInterval) {
+    return null;
+  }
+
+  useEffect(() => {
+    // If there are new changes, mark autoSave as not done.
+    if (hasNewChanges && didAutoSave) {
+      setDidAutoSave(false);
```

Comment:
This is slightly recursive, but I also not sure exactly what we need the `setDidAutoSave` for. Let's assume the following steps:

1. First change is made. The effect below schedules the "save" in N seconds.
2. Second change is made. The effect below is ignored.
3. Save timeout callback executes. Both `didAutoSave` and `hasNewChanges` values change (`true` and `false` respectively). The `didAutoSave` stays `true`.
4. Third change is made. The `didAutoSave` is set to `false`. The scheduling effect is executed again.

Here, I have to questions:
1. It seems that `didAutoSave` just follows `hasNewChanges` pretty closely. But with a 1-render-phase behind. Not super likely that this could cause race conditions. But still seems like it's not needed.
2. In the step 2 above, is this the desired behavior? Was your intention to keep the previously scheduled auto-save? Or reschedule it?

Reply:
Oh, I think this is actually be a left-over from supporting auto-save for published posts -- in case of published posts `hasNewChanges` wasn't reliable since that tracks the changes compared to the "real" saved story.

Will remove this, assuming we're not supporting auto-saving for published posts for now.

Reply:
All the logic for published posts should be removed now, also added tests now, PTAL.